src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+    int res = aNSC_CHECK_BUY_NORMAL_SUB;
+
+    for (i = 0; i < mPr_POCKETS_SLOT_COUNT; i++) {
+        if (*item == itm_name) {
+            if (!((NowPrivate_GetItemCond(i) & mPr_ITEM_COND_QUEST) || (NowPrivate_GetItemCond(i) & mPr_ITEM_COND_PRESENT))) {
+                *quant += 1;
+            }
+        }
+        item += 1;
+    }
+    if (*quant > 1) {
+        aNSC_set_buy_sum_str(itm_name, *quant);
+        res = aNSC_CHECK_BUY_OFFER_BUY_ALL;
+    }
+    return res;
+}
+
+static int aNSC_check_buy_paper(u32 *quant, mActor_name_t itm_name) {
+    int i;
+    int res = 0; 
+    mActor_name_t *pockets = Now_Private->inventory.pockets;
+    int paper_type = PAPER2TYPE(itm_name - ITM_PAPER_START);
+
+    for (i = 0; i < mPr_POCKETS_SLOT_COUNT; i++) {
+        mActor_name_t item = *pockets;
+        if (ITEM_IS_PAPER(item)) {
+            if (PAPER2TYPE(item - ITM_PAPER_START) == paper_type) {
+                if (!((NowPrivate_GetItemCond(i) & mPr_ITEM_COND_QUEST) || (NowPrivate_GetItemCond(i) & mPr_ITEM_COND_PRESENT))) {
+                    *quant += PAPER2STACK(item - ITM_PAPER_START) + 1;
+                }
+            }
+        }
+        pockets++;
+    }
+    if (*quant > 1) {
+        aNSC_set_buy_sum_str(itm_name, *quant);
+        res = aNSC_CHECK_BUY_OFFER_BUY_ALL;
+    }
+    return res;
+}
+
+static int aNSC_get_msg_no(int msg_no) {
+    int res = 0x82a;
+    if (msg_no < 0) {
+        return res;
+    }
+    else if (msg_no < 100) {
+        res = msg_no + 0x107b;
+    }
+    else if (msg_no < 200) {
+        res = msg_no + 0x2baf;
+    }
+    else if (msg_no < 300) {
+        res = msg_no + 0x2de6;
+    } else {
+        res = msg_no + 0x3cd2;
+    }
+    return res;
+}
+
+static void aNSC_Set_continue_msg_num(mMsg_Window_c* msg_p, NPC_SHOP_MASTER_ACTOR* shop_master, int msg_no) {
+    mMsg_Set_continue_msg_num(msg_p, msg_no);
+    shop_master->_9C4 = msg_no;
+}
+
+static void aNSC_ChangeMsgData(mMsg_Window_c* msg_p, NPC_SHOP_MASTER_ACTOR* shop_master, int p3) {
+    mMsg_ChangeMsgData(msg_p, p3);
+    shop_master->_9C4 = p3;
+}
+
+static void aNSC_set_player_angle()  {
+    static f32 fc_1, fc_2, fc_3;
+    PLAYER_ACTOR *player_actor;
+    s_xyz pos_8;
+    xyz_t stack_1;
+
+    player_actor = GET_PLAYER_ACTOR_NOW();
+
+    stack_1.x = ((f32)((int)player_actor->forward_ut_pos.x / mFI_UT_WORLDSIZE_X) + 0.5f) * mFI_UT_WORLDSIZE_X_F;
+    stack_1.z = ((f32)((int)player_actor->forward_ut_pos.z / mFI_UT_WORLDSIZE_Z) + 0.5f) * mFI_UT_WORLDSIZE_Z_F;
+
+    pos_8.y = atans_table(stack_1.z - player_actor->actor_class.world.position.z, stack_1.x - player_actor->actor_class.world.position.x);
+
+    GET_PLAYER_ACTOR_NOW()->Set_force_position_angle_proc(gamePT, &player_actor->actor_class.world.position, &pos_8, mPlayer_FORCE_POSITION_ANGLE_ROTY);
+}
+
+static void aNSC_set_talk_info_show_item(NPC_SHOP_MASTER_ACTOR *actor, mActor_name_t item) {
+    aNSC_set_item_str(actor, item);
+    aNSC_set_player_angle();
+    mDemo_Set_camera(1);
```

Comment:
This has an enum associated with it:
`CAMERA2_PROCESS_*`

This one is `CAMERA2_PROCESS_NORMAL`.